flink-table-store-connector/src/main/java/org/apache/flink/table/store/connector/sink/global/LocalCommitterOperator.java

```diff
@@ -0,0 +1,205 @@
+
+import static org.apache.flink.util.Preconditions.checkNotNull;
+
+/** A {@link AbstractCommitterOperator} to process local committer. */
+public class LocalCommitterOperator<CommT> extends AbstractCommitterOperator<CommT, CommT> {
+
+    private static final long serialVersionUID = 1L;
+
+    private final SerializableSupplier<Committer<CommT>> committerFactory;
+
+    private Committer<CommT> committer;
+
+    public LocalCommitterOperator(
+            SerializableSupplier<Committer<CommT>> committerFactory,
+            SerializableSupplier<SimpleVersionedSerializer<CommT>> committableSerializer) {
+        super(committableSerializer);
+        this.committerFactory = checkNotNull(committerFactory);
+    }
+
+    @Override
+    public void initializeState(StateInitializationContext context) throws Exception {
+        committer = committerFactory.get();
+        super.initializeState(context);
+    }
+
+    @Override
+    public void commit(boolean isRecover, List<CommT> committables)
+            throws IOException, InterruptedException {
+        if (committables.isEmpty()) {
+            return;
+        }
+
+        List<CommitRequestImpl> requests = new ArrayList<>(committables.size());
+        for (CommT comm : committables) {
+            requests.add(new CommitRequestImpl(comm));
+        }
+
+        long sleep = 1000;
+        while (true) {
+            // commit
+            requests.forEach(CommitRequestImpl::setSelected);
+            committer.commit(new ArrayList<>(requests));
+            requests.forEach(CommitRequestImpl::setCommittedIfNoError);
+
+            // drain finished
+            requests.removeIf(CommitRequestImpl::isFinished);
+            if (requests.isEmpty()) {
+                return;
+            }
+
+            //noinspection BusyWait
+            Thread.sleep(sleep);
+            sleep *= 2;
+        }
+    }
+
+    @Override
+    public List<CommT> toCommittables(long checkpoint, List<CommT> inputs) {
+        return inputs;
+    }
+
+    @Override
+    public void close() throws Exception {
+        committer.close();
+        super.close();
+    }
+
+    /** {@link CommitRequest} implementation. */
+    public class CommitRequestImpl implements CommitRequest<CommT> {
+
+        private CommT committable;
+        private int numRetries;
+        private CommitRequestState state;
+
+        private CommitRequestImpl(CommT committable) {
+            this.committable = committable;
+            this.state = CommitRequestState.RECEIVED;
+        }
+
+        private boolean isFinished() {
+            return state.isFinalState();
+        }
+
+        @Override
+        public CommT getCommittable() {
+            return this.committable;
+        }
+
+        @Override
+        public int getNumberOfRetries() {
+            return this.numRetries;
+        }
+
+        @Override
+        public void signalFailedWithKnownReason(Throwable t) {
+            this.state = CommitRequestState.FAILED;
```

Comment:
Should we throw an exception here? O.w. the exception will be swallowed without any trace.